Hunting Queries/MultipleDataSources/PermutationsOnLogonNames.yaml

```diff
@@ -0,0 +1,115 @@
+  | project TimeGenerated, CharSet, UserNameMatchOnFirst, UserNameMatchOnFirstCount = array_length(UserNameMatchOnFirst), FailedLogonCountForFirst, Type; 
+  let Last6charList = Char6List | where isnotempty(Last6char) | summarize UserNameMatchOnLast = makeset(UserNames), FailedLogonCountForLast = sum(FailedLogonCount) by TimeGenerated, CharSet = Last6char, Type
+  | project TimeGenerated, CharSet, UserNameMatchOnLast, UserNameMatchOnLastCount = array_length(UserNameMatchOnLast), FailedLogonCountForLast, Type; 
+  let char6 = First6charList | join ( 
+  Last6charList  
+  ) on $left.CharSet == $right.CharSet and $left.TimeGenerated == $right.TimeGenerated 
+  | project-away TimeGenerated1, CharSet1
+  // Make sure that we get more than a single match for First or Last
+  | where UserNameMatchOnFirstCount >= UserNameMatchMin or UserNameMatchOnLastCount >= UserNameMatchMin
+  | where FailedLogonCountForFirst >= FailedLogonMin or FailedLogonCountForLast >= FailedLogonMin
+  ; 
+  let Char8List = UserNameList  
+  | project TimeGenerated, First8char, Last8char, UserNames, FailedLogonCount, Type 
+  | where First8char != "LessThan8" and Last8char != "LessThan8"; 
+  // Break out first and last so we can then join and see where a first and last match. 
+  let First8charList = Char8List | where isnotempty(First8char) | summarize UserNameMatchOnFirst = makeset(UserNames), FailedLogonCountForFirst = sum(FailedLogonCount) by TimeGenerated, CharSet = First8char, Type
+  | project TimeGenerated, CharSet, UserNameMatchOnFirst, UserNameMatchOnFirstCount = array_length(UserNameMatchOnFirst), FailedLogonCountForFirst, Type; 
+  let Last8charList = Char8List | where isnotempty(Last8char) | summarize UserNameMatchOnLast = makeset(UserNames), FailedLogonCountForLast = sum(FailedLogonCount) by TimeGenerated, CharSet = Last8char, Type
+  | project TimeGenerated, CharSet, UserNameMatchOnLast, UserNameMatchOnLastCount = array_length(UserNameMatchOnLast), FailedLogonCountForLast, Type; 
+  let char8 = First8charList | join ( 
+  Last8charList  
+  ) on $left.CharSet == $right.CharSet and $left.TimeGenerated == $right.TimeGenerated 
+  | project-away TimeGenerated1, CharSet1 
+  // Make sure that we get more than a single match for First or Last
+  | where UserNameMatchOnFirstCount >= UserNameMatchMin or UserNameMatchOnLastCount >= UserNameMatchMin
+  | where FailedLogonCountForFirst >= FailedLogonMin or FailedLogonCountForLast >= FailedLogonMin
+  ; 
+  (union isfuzzy=true char4, char6, char8) 
+  | project Type, TimeGenerated, CharSet, UserNameMatchOnFirst, UserNameMatchOnFirstCount, FailedLogonCountForFirst, UserNameMatchOnLast, UserNameMatchOnLastCount, FailedLogonCountForLast 
+  | sort by UserNameMatchOnFirstCount desc, UserNameMatchOnLastCount desc
+  | extend timestamp = TimeGenerated
```

Comment:
Maybe add AccountCustomEntity for the target account(s?)?

Reply:
This is a complex query which I happened to write, I don't want to mess with it as part of these changes.  I will see if we can do this in the future.